tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/FilterStaticFrameworksTaskTest.cs

```diff
@@ -0,0 +1,241 @@
+
+namespace Xamarin.MacDev.Tasks.Tests {
+
+	[TestFixture]
+	public class FilterStaticFrameworksTaskTest {
+
+		string tempDir = "";
+
+		[SetUp]
+		public void Setup ()
+		{
+			tempDir = Cache.CreateTemporaryDirectory ();
+		}
+
+		[TearDown]
+		public void TearDown ()
+		{
+			if (Directory.Exists (tempDir))
+				Directory.Delete (tempDir, true);
+		}
+
+		[Test]
+		public void TestCustomFrameworkExecutablePath_iOS ()
+		{
+			TestCustomFrameworkExecutablePathForPlatform ("iOS", "libavcodec.framework", "libavcodec.dylib", false);
+		}
+
+		[Test]
+		public void TestCustomFrameworkExecutablePath_tvOS ()
+		{
+			TestCustomFrameworkExecutablePathForPlatform ("tvOS", "libavcodec.framework", "libavcodec.dylib", false);
+		}
+
+		[Test]
+		public void TestCustomFrameworkExecutablePath_macOS ()
+		{
+			TestCustomFrameworkExecutablePathForPlatform ("macOS", "libavcodec.framework", "libavcodec.dylib", true);
+		}
+
+		[Test]
+		public void TestCustomFrameworkExecutablePath_MacCatalyst ()
+		{
+			TestCustomFrameworkExecutablePathForPlatform ("MacCatalyst", "libavcodec.framework", "libavcodec.dylib", true);
+		}
+
+		void TestCustomFrameworkExecutablePathForPlatform (string platform, string frameworkName, string executableName, bool usesVersionsStructure)
+		{
+			// Arrange: Create a mock framework with custom CFBundleExecutable
+			var frameworkDir = Path.Combine (tempDir, platform, frameworkName);
+			Directory.CreateDirectory (frameworkDir);
+
+			string infoPlistPath;
+			if (usesVersionsStructure) {
+				// macOS and MacCatalyst structure: Framework.framework/Versions/A/Resources/Info.plist
+				var versionsDir = Path.Combine (frameworkDir, "Versions", "A");
+				var resourcesDir = Path.Combine (versionsDir, "Resources");
+				Directory.CreateDirectory (resourcesDir);
+				infoPlistPath = Path.Combine (resourcesDir, "Info.plist");
+				
+				// Create symlinks as they exist in real frameworks
+				if (!Directory.Exists (Path.Combine (frameworkDir, "Resources"))) {
+					Directory.CreateSymbolicLink (Path.Combine (frameworkDir, "Resources"), "Versions/A/Resources");
+				}
+			} else {
+				// iOS and tvOS structure: Framework.framework/Info.plist
+				infoPlistPath = Path.Combine (frameworkDir, "Info.plist");
+			}
+
+			// Create Info.plist with custom CFBundleExecutable
+			var infoPlistContent = @"<?xml version=""1.0"" encoding=""UTF-8""?>
+<!DOCTYPE plist PUBLIC ""-//Apple//DTD PLIST 1.0//EN"" ""http://www.apple.com/DTDs/PropertyList-1.0.dtd"">
+<plist version=""1.0"">
+<dict>
+	<key>CFBundleExecutable</key>
+	<string>" + executableName + @"</string>
+	<key>CFBundleIdentifier</key>
+	<string>com.ffmpeg.libavcodec</string>
+</dict>
+</plist>";
+			File.WriteAllText (infoPlistPath, infoPlistContent);
+
+			// Create the custom executable file 
+			var customExecutablePath = Path.Combine (frameworkDir, executableName);
+			File.WriteAllText (customExecutablePath, "mock executable");
+
+			// Act: Use reflection to test the helper method
```

Comment:
Instead of using reflection: create an instance of the FilterStaticFrameworks task, set the correct properties, execute the task, and verify that no errors were shown and the filtered frameworks.